docs/style-guidelines.md

```diff
@@ -1,20 +1,147 @@
+  * Nested classes
+### Order inside group:
+  * `public` 
+  * `public` with `[PublicApi]` and `[UsedImplicitly]` attributes
+  * `internal`
+  * `protected internal`
+  * `protected`
+  * `private`
+  * methods has additional rule for ordering
+### Fields specifics
+  * `Const`, `static` and `readonly` goes first (see example below 👇)
+  * `public` fields with `[HideInInspector]` and `[NonSerialized]` attribute goes after `public`'s
+  * `[SerializedFields]` attribute fields goes after all `public` fields and before `internal`-`private`-`protected`)
+```csharp
+public const string ASSET_PATH = "AvatarPrefab";                              // Constants  
+internal const float MIN_TIME = 0.1f;
+
+public static readonly int WAVE_ANIM_HASH = Animator.StringToHash("Wave");    // Static Read-only 
+private static readonly int DANCE_ANIM_HASH = Animator.StringToHash("Dance");
+
+public static bool IsOpen;                                                    // Static 
+internal static bool isAvtive;                                             
+
+public readonly List<int> CachedNumbers = new List<int>();                    // Read-only 
+protected readonly List<Section> sections = new List<Section>();
+
+public int PublicFieldA;                                                      // Public  
+[HideInInspector] public int SomeField02;
+[NonSerialized] public int SomeField03;
+
+[SerializedField] internal Animator animator;                                 // [SerializedField]'s 
+[SerializedField] private AnimationClip[] clips;  
+
+protected float cooldown;                                                     // internal-protected-private's
+private bool isVisible;                         
+```
+### Properties specifics
+  * static and readonly goes first (similar as for fields, see example above 👆)
+  * `get`-only and `set`-only properties goes last.
+  * Access modifiers for `set` considered to have higher priority than `get` (considered to be more exposed). 
+    * For example, property with `public set` and `private get` goes before `get`-only Property or Property with `private set` and `public get`
+    
+```csharp
+public bool Property1 { get; set; }                                        // get-set order
+public bool Property2 { private get; set; }
+public bool Property3 { get; private set; }
+public bool Property4 { get; }
+
+protected bool Property5 { get; set; }
+private bool Property6 { get; set; }
+```
+### **Methods specifics**
+  * Helper and supplementary methods which called from another method should be placed after method that calls it (in most cases it is `private` functions).
+```csharp
+// Note: indentaion for helper methods is used only for clarity. It is not a part of our formating style 
+public void Test1()                   // called by other class
+{
+    Test1Helper1();
+    Test1Helper2();
+}
+
+  private void Test1Helper1() => A();  // called by Test1() 
+    internal void A() { }              // called by Test1Helper1()
```

Comment:
What about using local functions for those types of invocations? 
```suggestion
  private void Test1Helper1()
  { 
     LocalA();
     void LocalA() { }
  };  // called by Test1() 
```

Reply:
agreed

Reply:
We can try to propose it. I was using this approach once, but it wasn't that good. Also, we there are some edge-cases for heap-allocation to keep in mind with local functions  https://devblogs.microsoft.com/premier-developer/dissecting-the-local-functions-in-c-7/

Reply:
@lorux0, yep, I think I mentioned in the comment that this indentation is not a part of our code style and only used for clarity in this example